src/ia-account-settings.ts

```diff
@@ -0,0 +1,897 @@
+
+  private oldUserData: UserModel = {
+    email: '',
+    screenname: '',
+  };
+
+  // we need to store the provider user wants to unlink from account
+  private unlinkProviders: LinkedProvidersModel = {};
+
+  firstUpdated() {
+    this.oldUserData = Object.assign(this.oldUserData, {
+      email: this.userData.email,
+      screenname: this.userData.screenname,
+    });
+
+    this.bindEvents();
+  }
+
+  updated(changed: PropertyValues) {
+    this.fileInput = this.iaPicUploader?.shadowRoot?.querySelector(
+      '.file-selector'
+    ) as HTMLInputElement;
+
+    if (changed.has('userData') || changed.has('selectedMailingLists')) {
+      this.showLoadingIndicator = false;
+      this.updatedFields = {};
+    }
+
+    if (changed.has('fieldsError') && this.hasFieldError()) {
+      log('buttton keep disabled');
+      this.saveButtonDisabled = true;
+    }
+  }
+
+  bindEvents() {
+    // check if google auth authentication completed, show setting page
+    document.addEventListener('IAThirdPartyAuth:verifiedLogin', () => {
+      log('IAThirdPartyAuth:verifiedLogin');
+      this.lookingToAuth = false;
+    });
+
+    // scroll event to make sticky header
+    window.addEventListener(
+      'scroll',
+      throttle(() => {
+        if (window.scrollY > this.offsetTop) {
+          this.isStickyHeader = true;
+          log('sticky header');
+        } else {
+          this.isStickyHeader = false;
+          log('non-sticky header');
+        }
+      }, 50)
+    );
+  }
+
+  render() {
+    return html`
+      <main id="maincontent">
+        <div class="container">
+          ${this.lookingToAuth
+            ? this.verificationTemplate
+            : this.settingsTemplate}
+        </div>
+      </main>
+    `;
+  }
+
+  /** @inheritdoc */
+  setScreenname(e: Event) {
+    const input = e.target as HTMLInputElement;
+    this.userData.screenname = input.value;
+    this.resetErrorFields('screenname');
+  }
+
+  /** @inheritdoc */
+  setEmail(e: Event) {
+    const input = e.target as HTMLInputElement;
+    this.userData.email = input.value;
+    this.fieldsError.email = '';
+    this.resetErrorFields('email');
+  }
+
+  /** @inheritdoc */
+  setPassword(e: Event) {
+    const input = e.target as HTMLInputElement;
+    this.userData.password = input.value;
+    this.fieldsError.password = '';
+    this.resetErrorFields('password');
+  }
+
+  /** @inheritdoc */
+  setBorrowHistory(e: Event) {
+    const input = e.target as HTMLInputElement;
+    this.loanHistoryFlag = input.checked ? 'public' : 'private';
+    this.changeSaveButtonState();
+  }
+
+  /** @inheritdoc */
+  setMailingList(e: Event) {
+    const input = e.target as HTMLInputElement;
+    const fieldName = input.name;
+
+    if (input.checked) {
+      this.selectedMailingLists.push(fieldName);
+    } else {
+      const index = this.selectedMailingLists.indexOf(fieldName);
+      this.selectedMailingLists.splice(index, 1);
+    }
```

Comment:
Q: I'm still a bit confused by the type of `selectedMailingLists`. It's initialized to an empty object `{}` above, but here it's having array methods called on it. Can we double-check this?